ui/apps/platform/src/Containers/Clusters/ClustersTable.tsx

```diff
@@ -0,0 +1,246 @@
+
+    function toggle(clusterId: string, col: ColId) {
+        setExpanded((prev) => ({
+            ...prev,
+            [clusterId]: prev[clusterId] === col ? null : col,
+        }));
+    }
+
+    function isCellExpanded(clusterId: string, col: ColId) {
+        return expanded[clusterId] === col;
+    }
+
+    function isRowExpanded(clusterId: string) {
+        return expanded[clusterId] != null;
+    }
+
+    function isHelmManaged(cluster: Cluster) {
+        return (
+            cluster.managedBy === 'MANAGER_TYPE_HELM_CHART' ||
+            (cluster.managedBy === 'MANAGER_TYPE_UNKNOWN' && !!cluster.helmConfig)
+        );
+    }
+
+    function isOperatorManaged(cluster: Cluster) {
+        return cluster.managedBy === 'MANAGER_TYPE_KUBERNETES_OPERATOR';
+    }
+
+    const colSpan = 9;
+
+    return (
+        <Table>
+            <Thead>
+                <Tr>
+                    <Th
+                        select={{
+                            onSelect: () => toggleAllClusters(),
+                            isSelected:
+                                tableState.type === 'COMPLETE' &&
+                                tableState.data.length === selectedClusterIds.length,
+                        }}
+                    />
+                    <Th>Cluster</Th>
+                    <Th>Provider (Region)</Th>
+                    <Th>Status</Th>
+                    <Th>Sensor upgrade status</Th>
+                    <Th>Credential expiration</Th>
+                    <Th>Cluster deletion</Th>
+                </Tr>
+            </Thead>
+            <TbodyUnified
+                tableState={tableState}
+                colSpan={colSpan}
+                errorProps={{
+                    title: 'There was an error loading cluster information',
+                }}
+                filteredEmptyProps={{ onClearFilters }}
+                renderer={({ data }) => (
+                    <>
+                        {data.map((clusterInfo, rowIndex) => {
+                            const provider = formatCloudProvider(
+                                clusterInfo.status?.providerMetadata
+                            );
+                            const clusterId = clusterInfo.id;
+                            return (
+                                <Tbody isExpanded={isRowExpanded(clusterId)} key={clusterInfo.id}>
+                                    <Tr key={clusterInfo.id}>
+                                        <Td
+                                            select={{
+                                                rowIndex,
+                                                onSelect: () => toggleCluster(clusterId),
+                                                isSelected: selectedClusterIds.includes(clusterId),
+                                            }}
+                                        />
+                                        <Td dataLabel="Cluster">
+                                            <Flex
```

Comment:
* One side of coin: Bravo for simple port from `clusterTableColumnDescriptors`
* Other side of coin: What are pro and con for potential follow up to factor out as small React component?

Reply:
Probably a good use case, especially while this code can still be shared with the "old" table.